api/groups/nodeGroup_test.go

```diff
@@ -387,6 +395,66 @@ func TestPeerInfo_PeerInfoShouldWork(t *testing.T) {
 	assert.NotNil(t, responseInfo["info"])
 }
 
+func TestEpochStartData_FacadeErrorsShouldErr(t *testing.T) {
+	t.Parallel()
+
+	expectedErr := errors.New("expected error")
+	facade := mock.FacadeStub{
+		GetEpochStartDataForEpochCalled: func(epoch uint32) (*common.EpochStartDataAPI, error) {
+			return nil, expectedErr
+		},
+	}
+
+	nodeGroup, err := groups.NewNodeGroup(&facade)
+	require.NoError(t, err)
+
+	ws := startWebServer(nodeGroup, "node", getNodeRoutesConfig())
+
+	req, _ := http.NewRequest("GET", "/node/epoch-start/4", nil)
+	resp := httptest.NewRecorder()
+	ws.ServeHTTP(resp, req)
+
+	response := &shared.GenericAPIResponse{}
+	loadResponse(resp.Body, response)
+
+	assert.Equal(t, http.StatusInternalServerError, resp.Code)
+	assert.True(t, strings.Contains(response.Error, expectedErr.Error()))
+}
+
+func TestEpochStartData_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	expectedEpochStartData := &common.EpochStartDataAPI{
+		Nonce:     1,
+		Round:     2,
+		ShardID:   3,
+		Timestamp: 4,
+	}
+
+	facade := mock.FacadeStub{
+		GetEpochStartDataForEpochCalled: func(epoch uint32) (*common.EpochStartDataAPI, error) {
+			return expectedEpochStartData, nil
+		},
+	}
+
+	nodeGroup, err := groups.NewNodeGroup(&facade)
+	require.NoError(t, err)
+
+	ws := startWebServer(nodeGroup, "node", getNodeRoutesConfig())
+
+	req, _ := http.NewRequest("GET", "/node/epoch-start/3", nil)
+	resp := httptest.NewRecorder()
+	ws.ServeHTTP(resp, req)
+
+	response := &epochStartResponse{}
+	loadResponse(resp.Body, response)
+
+	assert.Equal(t, http.StatusOK, resp.Code)
+	assert.Equal(t, "", response.Error)
+
+	require.Equal(t, *expectedEpochStartData, response.Data.EpochStartDataAPI)
```

Comment:
Perhaps `GetEpochStartDataForEpoch` (the production code) can be adjusted to return the DTO, instead of a pointer (optional).

Reply:
I think it's easier to work with it as a pointer (I can return `nil, err` in case of error, instead of `EpochStartData{}, err`)